rattr/plugins/analysers/builtins.py

```diff
@@ -2,14 +2,43 @@
 from __future__ import annotations
 
 import ast
+from typing import TYPE_CHECKING, NamedTuple
 
 from rattr.analyser.base import CustomFunctionAnalyser
 from rattr.analyser.util import get_dynamic_name
 from rattr.ast.util import fullname_of
 from rattr.models.context import Context
-from rattr.models.ir import FunctionIr
 from rattr.models.symbol import Name
 
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    from rattr.ast.types import Identifier
+    from rattr.models.ir import FunctionIr
+
+
+class AccessedAttributes(NamedTuple):
+    full: Name
+    lhs_names: list[Name]
+
+
+def iter_lhs_names(target: Name) -> Iterator[Identifier]:
+    parts = target.name.split(".")
+
+    if parts == [target]:
+        yield target
+        return
+
+    for end_offset in range(1, len(parts)):
+        yield ".".join(parts[:-end_offset])
```

Comment:
`foo.bar.baz.attr` should return `['foo.bar.baz', 'foo.bar', 'foo']` because we want the LHS names not the full name. Because when we access `foo.bar.baz.attr` we already know that `foo.bar.baz.attr` itself is accessed but we also want `foo.bar.baz` etc to be marked as accessed.